olive/engine/engine.py

```diff
@@ -85,16 +85,13 @@ def __init__(
         if execution_providers is None:
             execution_providers = self._config.execution_providers
 
-        # verify the AzureML system have specified the execution providers
-        # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
-        if self.target.system_type == SystemType.AzureML and execution_providers is None:
-            raise ValueError("AzureMLSystem requires execution providers to be specified.")
+        # for managed environment, the user must specify the execution providers
+        if self.target.olive_managed_env and execution_providers is None:
+            raise ValueError("Managed environment requires execution providers to be specified.")
```

Comment:
Is another main idea of `olive_managed_env` that user doesn't need to worry about ep and we will try all possible eps for them? I think that was why we skip get_supported_execution_providers call for `olive_managed_env`.

Reply:
In this PR I didn't want to change any of the original behavior. Since Mike already merged his PR to correct the accelerator spec resolution, this change doesn't exist in this PR anymore.